kernel/src/main/scala/cats/kernel/instances/SeqInstances.scala

```diff
@@ -83,7 +83,9 @@ class SeqMonoid[A] extends Monoid[Seq[A]] {
 }
 
 object SeqMonoid {
```

Comment:
I wonder if we want to not just deprecate `SeqMonoid` but also make it `private [cats]`:
https://github.com/typelevel/cats/blob/17d7e95e0b0bbc8b5763a2b2904fdb807dcea5b1/kernel/src/main/scala/cats/kernel/instances/SeqInstances.scala#L73-L74

Looks like `SeqMonoid` is not supposed to be user-facing, so it should be a fairly safe change.
It would be out of scope of this PR, of course, but if it makes sense, I can do it here, because it is still related to the warnings cause.

Reply:
That'd be a source breaking change for whoever used it that way.  Even though they _shouldn't_ use it, it's nice to make it very clear how to fix it.  I don't tend to like to remove deprecations until keeping them causes pain.

Reply:
I feel the best strategy would here be to provide a scalafix rule for that. But it would definitely come out of the scope of this PR.